modules/querier/worker/frontend_processor.go

```diff
@@ -66,8 +68,10 @@ func (fp *frontendProcessor) processQueriesOnSingleStream(ctx context.Context, c
 		}
 
 		if err := fp.process(c); err != nil {
-			level.Error(fp.log).Log("msg", "error processing requests", "address", address, "err", err)
-			backoff.Wait()
+			if status.Code(err) != codes.Canceled {
```

Comment:
So this is really interesting and worth a comment for sure. I didn't notice this behavior until I dug for the log message below.

It seems that when you cancel a query in Grafana the cancel is somehow propagated to the querier? B/c this line is hitting? i like dropping the backoff as well in this case b/c it will cause our queriers to reconnect quicker in the event of quick cancellations. 

further evidence this area is in need of overall improvement, but i think this is a great standalone patch.

Reply:
I've added a comment.